src/React/Basic.purs

```diff
@@ -38,20 +40,48 @@ foreign import data ReactComponent :: Type -> Type
 -- | The rendering function should return a value of type `JSX`, which can be
 -- | constructed using the helper functions provided by the `React.Basic.DOM`
 -- | module (and re-exported here).
-react
-  :: forall props state fx
+component
+  :: forall props state
    . { displayName :: String
      , initialState :: { | state }
-     , receiveProps :: { | props } -> { | state } -> (SetState state fx) -> Effect Unit
-     , render :: { | props } -> { | state } -> (SetState state fx) -> JSX
+     , receiveProps ::
+        { isFirstMount :: Boolean
```

Comment:
Might be worth pulling out a type synonym here?

Reply:
Yeah.. I couldn't decide. It's only repeated once and it seemed harder to read that way, plus naming the alias